### PR TITLE
Add test for decimal type

### PR DESCRIPTION
**Pull request by tadelesh**
# Cadl Ranch Contribution Checklist:

- [x] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [x] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [x] I have written a [mock API](../docs/writing-mock-apis.md)
- [x] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

### CODE REVIEW

**Comment by tadelesh** [comment]
@weidongxu-microsoft @chunyu3 @msyyc @qiaozha could you help to review this PR?

File: packages/cadl-ranch-specs/cadl-ranch-summary.md [hunk 1]
```diff
@@ -2775,6 +2775,60 @@ Expected Array input body:
...
+```json
+0.33333
+```
+
+### Type_Basic_Decimal128Type_requestParameter
+
+- Endpoint: `get /type/basic/decimal128/request_parameter`
+
+Expected request parameter:
+value=dGVzdA (base64url encode of test)
```
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/cadl-ranch-summary.md:2793]: ???

File: packages/cadl-ranch-specs/http/type/basic/mockapi.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+function createModelMockApis(route: string, value: any): MockApiOperations {
+  return {
+    responseBody: mockapi.get(`/type/basic/${route}/response_body`, (req) => {
+      return {
+        status: 200,
+        body: json(value),
+      };
+    }),
+    requestBody: mockapi.put(`/type/basic/${route}/resquest_body`, (req) => {
+      req.expect.coercedBodyEquals(value);
```
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/type/basic/mockapi.ts:20]: What does `coercedBodyEquals` do, compared with normal equals?

**Review by weidongxu-microsoft — COMMENTED** [review]
I think we already had scalar folder
https://github.com/Azure/cadl-ranch/tree/main/packages/cadl-ranch-specs/http/type/scalar

**Comment by tadelesh** [comment]
> I think we already had scalar folder https://github.com/Azure/cadl-ranch/tree/main/packages/cadl-ranch-specs/http/type/scalar

Moved to scalar. But I did not change the previous scalar types' test to prevent any breakings.

**Review by weidongxu-microsoft — APPROVED** [review]

File: packages/cadl-ranch-specs/cadl-ranch-summary.md [hunk 3]
```diff
@@ -4747,6 +4747,46 @@ Expected input body:
...
+```
+
+### Type_Property_ValueTypes_Decimal128_put
+
+- Endpoint: `put /type/property/value-types/decimal128`
+
+Expected input body:
+
+```json
+{ "property": 0.33333 }
```
Comment (msyyc) [inline packages/cadl-ranch-specs/cadl-ranch-summary.md:4787]: Don't know much about decimal. Could we test the boundary value?

Comment (timotheeguerin) [reply packages/cadl-ranch-specs/cadl-ranch-summary.md:4787]: we don't really have any test like that but maybe we could have something like

```tsp
op getDecimals(): { a: decimal128, b: decimal128, c: decimal128 };
op verifySum(@body sum: decimal128): void;
```

and the client test need to call `getDecimals()` then do the sum and upload it back to `verifySum`. If you are not using decimal and float, doubles you would send back the wrong value.

For example in JS(with double) if we sum `0.1` 3 times then you don't end up with `0.3` exactly
image

Comment (tadelesh) [reply packages/cadl-ranch-specs/cadl-ranch-summary.md:4837]: Added.

**Review by chunyu3 — APPROVED** [review]

**Review by qiaozha — APPROVED** [review]
